### PR TITLE
Add support for tuples in the form of anonymous structs

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This corresponds to https://github.com/mozilla/jsonschema-transpiler/pull/89. In addition to struct support, this pads `addititional_properties` with null values if there are any values within repeated records. This allows correspondence between additional properties and the index of the record it belongs to.

### CODE REVIEW

File: ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java [hunk 1]
```diff
@@ -323,13 +344,19 @@ private void processField(String jsonFieldName, Field field, Object val,
...
         Map<String, Object> props = additionalProperties == null ? null : new HashMap<>();
         transformForBqSchema(record, field.getSubFields(), props);
         if (props != null && !props.isEmpty()) {
-          additionalProperties.put(name, props);
+          repeatedAdditionalProperties.add(props);
+        } else {
+          repeatedAdditionalProperties.add(null);
         }
       });
+      if (!repeatedAdditionalProperties.stream().allMatch(p -> p == null)) {
```
Comment (jklukas) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java]:
```suggestion
      if (!repeatedAdditionalProperties.stream().allMatch(Objects::isNull)) {
```

Slight preference for a method reference here.

Comment (jklukas) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java]: And this approach to padding with nulls to preserve index of additional properties is very nice.

**Comment by acmiyaguchi** [comment]
I'm running into the following issue with repeated tuples in the telemetry.event.4 table:

`      "message" : "Error while reading data, error message: JSON parsing error in row starting at position 0: Nested arrays not allowed.",`

**Comment by acmiyaguchi** [comment]
A successful job run can be found here: https://console.cloud.google.com/dataflow/jobsDetail/locations/us-central1/jobs/2019-10-01_15_59_09-10409615498354741939?project=amiyaguchi-dev

In a similar fashion to https://github.com/mozilla/jsonschema-transpiler/pull/90#issuecomment-537224923, I've validated correctness on sampled data via the integration testing workflow. 

### Sum of stack traces in untrusted-modules
```sql
SELECT
  SUM(list.f0_),
  SUM(list.f1_)
FROM
  test_ingestion.telemetry__untrusted_modules_v4,
  UNNEST(payload.combined_stacks.stacks) AS stacks,
  UNNEST(stacks.list) AS list
```


Row | f0_ | f1_ |  
-- | -- | -- | --
1 | 68708.0 | 2.158269056801035E21 |  

```bash
cat landfill-integration.ndjson | \
jq -rc 'select(.attributeMap.document_type=="untrustedModules") | .payload | @base64d | fromjson.payload.combinedStacks.stacks | .[] | .[] | join(",")' | \
python3 -c "import sys; x=[tuple(map(float, x.split(','))) for x in sys.stdin.readlines()]; print(list(map(sum, zip(*x))))"

[68708.0, 2.1582690568010512e+21]
```

### Count of top 10 events types
```sql
SELECT
  parent.f1_,
  parent.f2_,
  COUNT(*)
FROM
  test_ingestion.telemetry__event_v4,
  UNNEST(payload.events.parent) parent
GROUP BY
  1,
  2
ORDER BY
  3 DESC
LIMIT
  10
```


Row | f1_ | f2_ | f0_ |  
-- | -- | -- | -- | --
1 | uptake.remotecontent.result | uptake | 8893 |  
2 | addonsManager | update | 755 |  
3 | addonsManager | uninstall | 296 |  
4 | addonsManager | action | 284 |  
5 | devtools.main | tool_timer | 221 |  
6 | addonsManager | install | 211 |  
7 | devtools.main | exit | 180 |  
8 | devtools.main | enter | 175 |  
9 | security.ui.certerror | load | 162 |  
10 | devtools.main | edit_rule | 150 |  

```bash
cat landfill-integration.ndjson | \
jq -rc 'select(.attributeMap.document_type=="event") | .payload | @base64d | fromjson.payload.events.parent | select(. != null) | .[] | [.[1], .[2]] | join("|")' | \
sort | uniq -c | sort -r | head -n10 | \
sed 's/^[[:space:]]*//g' | awk '{printf "%s|%s\n", $2,$1}'
```

f1_ | f2_ | f0_
-|-|-
uptake.remotecontent.result|uptake|8893
addonsManager|update|755
addonsManager|uninstall|296
addonsManager|action|284
devtools.main|tool_timer|221
addonsManager|install|211
devtools.main|exit|180
devtools.main|enter|175
security.ui.certerror|load|162
devtools.main|edit_rule|150

**Pull request merged**